### PR TITLE
DB-10327 Add dbId to SchemaWatcher 

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by ascend1 — APPROVED** [review]

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceSchemaWatcher.java [hunk 1]
```diff
@@ -35,21 +35,23 @@
     private static CleanupMode mode = CleanupMode.UNDEF;
     private static Semaphore sync;
 
+    public String dbName;
     public String schemaName;
     protected String userName;
 
-    public SpliceSchemaWatcher(String schemaName) {
+    public SpliceSchemaWatcher(String dbName, String schemaName) {
```
Comment (yxia92) [inline splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceSchemaWatcher.java:42]: Can we keep the original constructor `SpliceSchemaWatcher(String schemaName` and let it call `SpliceSchemaWatcher(null, schemaName)`? Maybe we can avoid touching so many ITs.

**Review by yxia92 — CHANGES_REQUESTED** [review]

File: splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceSchemaWatcher.java (hunk 1 above)
Comment (dgomezferro) [reply splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceSchemaWatcher.java:42]: +1, I was going to propose the same thing

Comment (arnaud-splice) [reply splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceSchemaWatcher.java:46]: Good idea, this PR is now much smaller!

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1585/

**Review by dgomezferro — APPROVED** [review]

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by yxia92 — APPROVED** [review]

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**